code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm

```diff
@@ -98,4 +95,160 @@
 		behavior.next_slash_buffed = FALSE
 
 	to_chat(xeno, SPAN_XENODANGER("You have waited too long, your slash will no longer deal increased damage!"))
-	button.icon_state = "template"
+
+
+// VAMPIRE LURKER
+
+/datum/action/xeno_action/activable/pounce/rush/additional_effects(mob/living/living_target) //pounce effects
+	var/mob/living/carbon/target = living_target
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	target.dazed += 5
+	xeno.flick_attack_overlay(target, "slash")   //fake slash to prevent disarm abuse
+	target.last_damage_data = create_cause_data(xeno.caste_type, xeno)
+	target.apply_armoured_damage(20, ARMOR_MELEE, BRUTE)
+	playsound(get_turf(target), 'sound/weapons/alien_claw_flesh3.ogg', 30, TRUE)
+	shake_camera(target, 2, 1)
+	apply_cooldown_override(40)
+
+/datum/action/xeno_action/activable/flurry/use_ability(atom/targeted_atom) //flurry ability
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	if (!istype(xeno))
+		return
+	if (!xeno.check_state())
+		return
+	if (!action_cooldown_check())
+		return
+
+	xeno.visible_message(SPAN_DANGER("[xeno] slashes frantically the area in front of it!"), \
+	SPAN_XENOWARNING("You unleash a barrage of slashes!"))
+	playsound(xeno, 'sound/effects/alien_tail_swipe2.ogg', 30)
+	apply_cooldown()
+
+	// Transient turf list
+	var/list/target_turfs = list()
+	var/list/temp_turfs = list()
+	var/list/telegraph_atom_list = list()
+
+	// Code to get a 1x3 area of turfs
+	var/turf/root = get_turf(xeno)
+	var/facing = get_dir(xeno, targeted_atom)
+	var/turf/infront = get_step(root, facing)
+	var/turf/infront_left = get_step(root, turn(facing, 45))
+	var/turf/infront_right = get_step(root, turn(facing, -45))
+	temp_turfs += infront
+	if (!(!infront || infront.density))
+		temp_turfs += infront_left
+	if (!(!infront || infront.density))
+		temp_turfs += infront_right
+
+	for (var/turf/current_turfs in temp_turfs)
+
+		if (!istype(current_turfs))
+			continue
+
+		if (current_turfs.density)
+			continue
+
+		target_turfs += current_turfs
+		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(current_turfs, 2)
+
+	for (var/turf/current_turfs in target_turfs)
+		for (var/mob/living/carbon/target in current_turfs)
+			if (target.stat == DEAD)
+				continue
+
+			if (!isXenoOrHuman(target) || xeno.can_not_harm(target))
+				continue
+
+			xeno.visible_message(SPAN_DANGER("[xeno] scratches [target] all around its body!"), \
+			SPAN_XENOWARNING("You slash [target] multiple times!"))
+			xeno.flick_attack_overlay(target, "slash")
+			target.last_damage_data = create_cause_data(xeno.caste_type, xeno)
+			log_attack("[key_name(xeno)] attacked [key_name(target)] with Flurry")
+			target.apply_armoured_damage(30, ARMOR_MELEE, BRUTE)
+			playsound(get_turf(target), 'sound/weapons/alien_claw_flesh4.ogg', 30, TRUE)
+			xeno.emote("roar")
+			xeno.flick_heal_overlay(1 SECONDS, "#00B800")
+			xeno.gain_health(30)
+			..()
+			return
```

Comment:
```suggestion
```
this early return means it'll just pick a random human in the 1x3 area. if this is intended just undo this but your PR desc talks about "target**s**"